python/ray/util/placement_group.py

```diff
@@ -80,6 +84,30 @@ def _get_none_zero_resource(self, bundle: List[Dict]):
                 return key, value
         assert False, "This code should be unreachable."
 
+    def _fill_bundle_cache_if_needed(self):
+        if not self.bundle_cache:
+            # Since creating placement group is async, it is
+            # possible table is not ready yet. To avoid the
+            # problem, we should keep trying with timeout.
+            timeout_second = 5
+            wait_interval = 0.05
+            timeout_cnt = 0
+            worker = ray.worker.global_worker
+            worker.check_connected()
+
+            while timeout_cnt < int(timeout_second / wait_interval):
+                pg_info = ray.state.state.placement_group_table(self.id)
+                if pg_info:
+                    self.bundle_cache = list(pg_info["bundles"].values())
+                    return
+                time.sleep(wait_interval)
+                timeout_cnt += 1
+
+            raise TimeoutError(
```

Comment:
RuntimeError? since this isn't a user timeout

Reply:
Didn't know about that. Will fix.